placementApp/models.py

```diff
@@ -36,6 +36,12 @@ class User(AbstractBaseUser):
     username = models.CharField(max_length=40, blank=True, null=True)
     f_name = models.CharField(max_length=20)
     l_name = models.CharField(max_length=20)
+    profile_image = models.ImageField(
+        default="profile_images/default.png",
+        upload_to="profile_images",
+        blank=True,
+        null=True,
+    )
```

Comment:
Check 154 line of models. It will not work as username is not being taken anywhere hence it is None which gives errors.
Change it to return self.student.f_name + " " +self.student.f_name + ", " + self.position.title   or add a str method in user model to save username.